### ISSUE TITLE
Screen reader doesn't focus on MenuPopover when Menu is expanded

### ISSUE THREAD

**Issue by ncphillips** [reach/reach-ui#777]
### Current Behavior

Screen reader doesn't focus on MenuPopover when Menu is expanded

### Expected behavior

Screen reader should focus on MenuPopover when Menu is expanded

### Reproducible example

[CodeSandbox Template](https://codesandbox.io/s/reach-ui-template-xonfs)

### Additional context

I'm not sure if it's a bug or a problem in how I'm using this library. Here is the source code:

```tsx
import React, { FC } from 'react'
import { Menu, MenuItems, MenuPopover } from '@reach/menu-button'
import styled from 'styled-components'
import css from '@styled-system/css'

export const LayoutMobile: FC<LayoutMobileProps> = ({ children }) => (
  <Menu>
    <MobileOnlyStack align='right' direction='horizontal' gap='4'>
      <MobileMenuButton />
    </MobileOnlyStack>

    <MobileMenuPopover>
      <ContentBlock maxWidth='300px'>
        <MobileMenuItems as={Stack} align='center' gap='10'>
          {children}
        </MobileMenuItems>
      </ContentBlock>
    </MobileMenuPopover>
  </Menu>
)

const MobileOnlyStack = styled(Stack)`
  ${respondTo.desktop`
    display: none;
  `}
`

const MobileMenuPopover = styled(MenuPopover)`
  position: fixed !important;
  width: 100%;
  height: calc(100vh - 48px);
  top: 48px !important;
  left: 0 !important;
  overflow: hidden;
  z-index: 1000;

  ${css({
    bg: 'gray.0',
    pt: 10,
    a: {
      fontWeight: 'bold',
    },
  })}

  ${respondTo.desktop`
    display: none;
  `}
`

const MobileMenuItems = styled(MenuItems)`
  outline: none;
`

const MobileMenuButton = styled(MenuButton)`
 // ....
`
```

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    |    menu-button     |        0.15.0  |
| React            |         |  17.0.1       |
| Browser          |    Chrome,Firefox,Safari     |         |
| Assistive tech   |    VoiceOver     |         |
| Operating System |     MacOS    |     11.2.3    |

**Comment by ncphillips** [comment]
Perhaps on a related note, when I run the Axe Dev Tools on my page with the @reach/menu-button expanded it fails the check:

<img width="886" alt="image" src="https://user-images.githubusercontent.com/824015/113909992-b6170e00-97ae-11eb-8268-51f8ec7f6df0.png">

**Comment by ncphillips** [comment]
I figured it out...sort of. It has something to do with the `as={Stack}`:
```
  <MobileMenuItems as={Stack} align='center' gap='10'>
```

When I remove that and just write my own styles for MobileMenuItem it seems to be fixed

**Comment by ncphillips** [comment]
I confirmed that removing the `as={Stack}` also fixed the Axe problems

I am not sure why this would be the case. 

This was definitely unexpected behaviour, but I don't know whether it's:
1. a bug inside @reach/menu-button
1. a bug with styled-components
1. just a funky circumstance

Feel free to close this issue, or convert into a different issue

**Comment by chaance** [comment]
Likely depends on your implementation of `Stack` as it may or may not spread some of the necessary DOM props used internally to provide these accessibility requirements. I'm assuming that `tabIndex` and `role` were not passed along to the underlying node. Glad you found a workaround 👍
